src/calculation/CompositeReportBuilder.ts

```diff
@@ -22,13 +22,32 @@ export class CompositeReportBuilder<T extends PopulationGroupResult> extends Abs
   report: CompositeMeasureReport;
   compositeScoringType: CompositeScoreType;
   compositeFraction: { numerator: number; denominator: number };
+  weightedComponents: Record<string, number>;
 
   constructor(compositeMeasure: fhir4.Measure, options: CalculationOptions) {
     super(compositeMeasure, options);
 
     this.compositeScoringType = getCompositeScoringFromMeasure(compositeMeasure) ?? 'all-or-nothing';
     this.compositeFraction = { numerator: 0, denominator: 0 };
     this.options = options;
+    this.weightedComponents = {};
+
+    // if the weight is not specified, then we assume the weight is 1
+    compositeMeasure.relatedArtifact?.forEach(ra => {
+      if (ra.resource && ra.type === 'composed-of') {
+        let weight = 1;
+        ra.extension?.find(extension => {
```

Comment:
Sorry, I didn't mean just changing the forEach to a find, I meant use `find` to actually find the matching extension and only do the valueDecimal access if you get a result back:

```ts
const weightExtension = ra.extension?.find(extension => ...);

if (weightExtension) // do stuff